coverage.py

```diff
@@ -535,57 +535,102 @@ def _input_identifier_types(cls):
+        log.info('FOUND %r' % existing_record)
+        return existing_record, was_registered
+
+    @classmethod
+    def bulk_register(cls, identifiers, data_source=None, collection=None,
+        force=False
+    ):
+        """Registers identifiers for future coverage.
 
-            log.info('FOUND %r' % existing_record)
-            return existing_record, was_registered
+        This method is primarily for use with CoverageProviders that use the
+        `registered_only` flag to process items. It's currently only in use
+        on the Metadata Wrangler.
 
-        was_registered = True
-        new_record, is_new = CoverageRecord.add_for(
-            identifier, source, operation=operation,
-            status=CoverageRecord.REGISTERED,
-            collection=collection
+        :param data_source: DataSource object or basestring representing a
+            DataSource name. If a basestring is given, the DataSource will be
+            autocreated.
+        :param collection: Collection object to be associated with the
+            CoverageRecords.
+        :param force: When True, even existing CoverageRecords will have
+            their status reset to CoverageRecord.REGISTERED.
+
+        :return: A tuple of two lists: the first has fresh new REGISTERED
+            CoverageRecords and the second list already has Identifiers that
+            were ignored because they already had coverage.
+
+        TODO: Take identifier eligibility into account when registering.
+        """
+        if not identifiers:
+            return list(), list()
+
+        _db = Session.object_session(identifiers[0])
+        data_source = cls._data_source_for_registration(_db, data_source)
+
+        if collection and cls.COVERAGE_COUNTS_FOR_EVERY_COLLECTION:
+            # There's no need for a collection on this CoverageRecord.
+            collection = None
+
+        new_records, ignored_identifiers = CoverageRecord.bulk_add(
+            identifiers, data_source, operation=cls.OPERATION,
+            status=CoverageRecord.REGISTERED, collection=collection,
+            force=force,
         )
-        log.info('CREATED %r' % new_record)
-        return new_record, was_registered
+
+        return new_records, ignored_identifiers
+
+    @classmethod
+    def _data_source_for_registration(cls, _db, data_source):
+        """Finds or creates a DataSource for the registration methods
+        `cls.register` and `cls.bulk_register`.
+        """
+        if not data_source:
+            return DataSource.lookup(_db, cls.DATA_SOURCE_NAME)
+        if isinstance(data_source, DataSource):
+            return data_source
+        if isinstance(data_source, basestring):
+            # If a string is passed, we assume it's meant to be a
+            # DataSource. Because on the metadata wrangler new Collection
+            # names may be passed, this ensures there will always be a
+            # DataSource.
```

Comment:
I think this should take `autocreate` as an argument so that the responsibility for passing in `True` lies with the metadata wrangler.